[quant][graphmode] Clean up and add more logging

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #40212 [quant][graphmode] Rename graph mode quantization API to `quantize_jit`
* #40198 [quant][api] Expose graph mode quantization API in `torch.quantization`
* #40197 [quant][graphmode] Add support for detach
* **#40196 [quant][graphmode] Clean up and add more logging**

Summary:
- separate passes in insert observers to make it more robust
- added print for quantization type
- added more logging for insert observers

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22106545](https://our.internmc.facebook.com/intern/diff/D22106545)

jerryzh168 committed:
[quant][graphmode] Clean up and add more logging

Summary:
- separate passes in insert observers to make it more robust
- added print for quantization type
- added more logging for insert observers

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vkuzo commented on torch/csrc/jit/passes/quantization/quantization_type.h:
```diff
@@ -1,11 +1,14 @@
 // Quantization type (dynamic quantization, static quantization).
 // Should match the Python enum in quantize_script.py
-enum class QuantType { DYNAMIC, STATIC };
+enum QuantType : uint8_t { DYNAMIC = 0, STATIC };
```
what's the motivation of removing enum class?

jerryzh168 replied:
had some compilation problem, I think we can do enum class and inherit from `uint8_t`

jerryzh168 replied:
this change is to make it more specific that DYNAMIC corresponds to 0 and STATIC corresponds to 1

vkuzo replied:
don't feel super strongly, but wouldn't it be good to explicitly not think of this as 0, 1, etc, and instead think of the logical value?

optional, feel free to ignore

vkuzo replied:
maybe this? https://stackoverflow.com/a/11421471

jerryzh168 replied:
since we cast from/to integer values I think it's better to make it more explicit about the correspondence

jerryzh168 committed:
Update on "[quant][graphmode] Clean up and add more logging"


Summary:
- separate passes in insert observers to make it more robust
- added print for quantization type
- added more logging for insert observers

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Clean up and add more logging"


Summary:
- separate passes in insert observers to make it more robust
- added print for quantization type
- added more logging for insert observers

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22106545](https://our.internmc.facebook.com/intern/diff/D22106545)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Clean up and add more logging"


Summary:
- separate passes in insert observers to make it more robust
- added print for quantization type
- added more logging for insert observers

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22106545](https://our.internmc.facebook.com/intern/diff/D22106545)

[ghstack-poisoned]
